Node.JS binding module not correctly downloaded in a Node.JS 19 environment

Easy test case to reproduce is to checkout https://github.com/matrix-org/matrix-appservice-irc cleanly twice, running `npm install` once with Node.JS 18 and one with Node.JS 19. Notice that the latter process seems to hang for a while before finishing. Afterwards, try running `node app.js` and you'll see a:

```
/tmp/matrix-appservice-irc/node_modules/@matrix-org/matrix-sdk-crypto-nodejs/index.js:234
    throw loadError
    ^

Error: Cannot find module '@matrix-org/matrix-sdk-crypto-nodejs-linux-x64-gnu'
Require stack:
```

Having tested this, it looks like the `node_modules/@matrix-org/matrix-sdk-crypto-nodejs/download-lib.js` script is very slow on Node 19 and after 2 minutes it fails to fetch the file.

#### Node 18
```
Downloading lib matrix-sdk-crypto.linux-x64-gnu.node from https://github.com/matrix-org/matrix-rust-sdk/releases/download/matrix-sdk-crypto-nodejs-v0.1.0-beta.1/matrix-sdk-crypto.linux-x64-gnu.node
Download Completed
node node_modules/@matrix-org/matrix-sdk-crypto-nodejs/download-lib.js  0.11s user 0.01s system 41% cpu 0.308 total
```

#### Node 19
```
Downloading lib matrix-sdk-crypto.linux-x64-gnu.node from https://github.com/matrix-org/matrix-rust-sdk/releases/download/matrix-sdk-crypto-nodejs-v0.1.0-beta.1/matrix-sdk-crypto.linux-x64-gnu.node
Download Completed
node node_modules/@matrix-org/matrix-sdk-crypto-nodejs/download-lib.js  0.13s user 0.02s system 0% cpu 2:00.29 total
```

--- a/.github/workflows/bindings_ci.yml
+++ b/.github/workflows/bindings_ci.yml
@@ -92,7 +92,7 @@ jobs:
       fail-fast: true
       matrix:
         os: [ubuntu-latest]
-        node-version: [14.0, 16.0, 18.0]
+        node-version: [14.0, 16.0, 18.0, 19.0]
         include:
           - os: ubuntu-latest
             os-name: 🐧

--- a/bindings/matrix-sdk-crypto-nodejs/download-lib.js
+++ b/bindings/matrix-sdk-crypto-nodejs/download-lib.js
@@ -1,6 +1,5 @@
 
 const { Agent } = require('https');
-const { whyIsNodeStillRunning } = require('why-is-node-still-running');
 const { DownloaderHelper } = require('node-downloader-helper');
 const { version } = require("./package.json");
 const { platform, arch } = process
@@ -34,8 +33,6 @@ function download_lib(libname) {
 
     dl.on('end', () => console.info('Download Completed'));
     dl.on('error', (err) => console.info('Download Failed', err));
-    dl.on('stateChanged', (err) => console.info('Download state', err));
-    dl.on('timeout', (err) => console.info('Download timeout', err));
     dl.on('progress', stats => {
         const progress = stats.progress.toFixed(1);
         const speed = byteHelper(stats.speed);
@@ -45,7 +42,7 @@ function download_lib(libname) {
         // print every one second (`progress.throttled` can be used instead)
         const currentTime = new Date();
         const elaspsedTime = currentTime - startTime;
-        if (elaspsedTime > 1000 || stats.progress === 100) {
+        if (elaspsedTime > 1000) {
             startTime = currentTime;
             console.info(`${speed}/s - ${progress}% [${downloaded}/${total}]`);
         }

--- a/bindings/matrix-sdk-crypto-nodejs/download-lib.js
+++ b/bindings/matrix-sdk-crypto-nodejs/download-lib.js
@@ -1,3 +1,6 @@
+
+const { Agent } = require('https');
+const { whyIsNodeStillRunning } = require('why-is-node-still-running');
 const { DownloaderHelper } = require('node-downloader-helper');
 const { version } = require("./package.json");
 const { platform, arch } = process
@@ -22,10 +25,17 @@ function download_lib(libname) {
     console.info(`Downloading lib ${libname} from ${url}`);
     const dl = new DownloaderHelper(url, __dirname, {
         override: true,
+        httpsRequestOptions: {
+          // Disable keepalive to prevent the process hanging open.
+          // https://github.com/matrix-org/matrix-rust-sdk/issues/1160
+          agent: new Agent({ keepAlive: false }),
+        },
     });
 
     dl.on('end', () => console.info('Download Completed'));
     dl.on('error', (err) => console.info('Download Failed', err));
+    dl.on('stateChanged', (err) => console.info('Download state', err));
+    dl.on('timeout', (err) => console.info('Download timeout', err));
     dl.on('progress', stats => {
         const progress = stats.progress.toFixed(1);
         const speed = byteHelper(stats.speed);
@@ -35,7 +45,7 @@ function download_lib(libname) {
         // print every one second (`progress.throttled` can be used instead)
         const currentTime = new Date();
         const elaspsedTime = currentTime - startTime;
-        if (elaspsedTime > 1000) {
+        if (elaspsedTime > 1000 || stats.progress === 100) {
             startTime = currentTime;
             console.info(`${speed}/s - ${progress}% [${downloaded}/${total}]`);
         }

--- a/bindings/matrix-sdk-crypto-nodejs/download-lib.js
+++ b/bindings/matrix-sdk-crypto-nodejs/download-lib.js
@@ -1,4 +1,3 @@
-
 const { Agent } = require('https');
 const { DownloaderHelper } = require('node-downloader-helper');
 const { version } = require("./package.json");

